content/learn/book/ecs/_index.md

```diff
@@ -6,4 +6,125 @@ template = "book-section.html"
+
+When we want to go beyond this tabular data storage, we can use **resources**: global singletons which store data in monolithic blobs.
+You might use resources to store one-off bits of state like the game's score, use it to interface with other libraries, or store secondary data structures like indexes to augment your use of entity-component data.
+
+In order to actually manipulate all of this data in interesting ways, we must use systems.
+**Systems** are Rust functions which request specific data from the world, as declared in their system parameters (function parameters): generally resources and entities that have a particular combination of components using queries.
+Systems are Rust functions that request data from the `World` (such as resources or entities with particular components) in order to perform tasks.
+All of the rules and behaviours of our game are governed by systems.
+Once the systems are added to our app the **scheduler** takes in this information and automatically runs our systems: typically once during each pass of the **game loop**.
+
+Bevy's scheduler is remarkably powerful: it uses the information about data access defined in our system parameters to automatically run systems in parallel.
+By default, every system in a stage runs in parallel with every other system in that stage (as long as threads exist to take the work): the only rule is that systems which have the ability to *write* to a particular piece of data (such as a resource) cannot be run at the same time as other systems which read or write to that same data.
+In Rust, this corresponds to mutable access, as declared by the use of `Query<&mut MyComponent>` or `ResMut<MyResource>`.
+
+On the next page, we'll create a simple "game" using the ECS so you can see how this all fits together.
+
+## Rust tips and tricks for Bevy's ECS
+
+If you're new to Rust, there are a few standard patterns and pitfalls that are particularly relevant to Bevy's ECS.
+Feel free to come back to this section later as you learn more about Bevy!
+
+### Tuple structs
+
+Tuple structs, defined like `struct Score(u64);` or `struct Position(u8, u8);`, are structs with unnamed fields, accessed by the order in which they were defined.
+`.0`, `.1` and so on are used to access fields, counting up from 0 in the order they were declared.
+
+`my_position.0 = 42` can seem opaque when you first read it, but in Rust it just means that you're assigning `42` to the first field of the tuple struct `my_position` (by convention, the x coordinate).
+Using ordinary structs with named fields rather than tuple structs (even for simple components and resources!) can go a long way to improving the clarity of your code by providing more descriptive field names.
+In this example, `my_position.x = 42` immediately communicates intent, rather than relying on the end user to remember the convention used.
```

Comment:
```suggestion
In this example, `my_position.x = 42` immediately communicates intent, rather than relying on future you to remember the convention used.
```
Minor nitpick: the end-user of a game usually doesn't look at the code.

Reply:
```suggestion
In this example, `my_position.x = 42` immediately communicates intent, rather than relying on future developers to remember the convention used.
```
I think it should be 'future developers' because more than 1 person may be working on the code.